Implementation of CAIP-2 for EOSIO blockchains (#33)
ChainAgnostic/CAIPs: CAIPs/caip-7.md

---
caip: 7
title: Blockchain Reference for the EOSIO Namespace
author: Sebastian Montero (@sebastianmontero)
discussions-to: https://github.com/ChainAgnostic/CAIPs/issues/32
status: Draft
type: Standard
created: 2020-11-30
updated: 2020-11-30
requires: 2
---

## Simple Summary

This document is about the details of the EOSIO namespaces and references for CAIP-2.

## Abstract

In CAIP-2 a general blockchain identification scheme is defined. This is the
implementation of CAIP-2 for EOSIO.

## Motivation

See CAIP-2.

## Specification

### EOSIO Namespace

> The namespace "EOSIO" refers to the EOSIO open-source blockchain platform.

Reviewer: ```suggestion
The namespace "eosio" refers to the EOSIO open-source blockchain platform.
```

#### Reference Definition

The definition is delegated to the [EOSIO Transactions Protocol Documentation](https://developers.eos.io/welcome/v2.0/protocol/transactions_protocol/#32-sign-transaction), the [Chain API Plugin Documentation of the EOSIO Developers Manual](https://developers.eos.io/manuals/eos/latest/nodeos/plugins/chain_api_plugin/api-reference/index?query=chain%20id&page=1#operation/get_info) and the pull request that implemented its generation [Chain ID generation implementation pull request](https://github.com/EOSIO/eos/pull/3425). 

> The Chain ID is the SHA256 hash of the genesis state of the chain, and its represented as hexadecimal number of 64 digits

Reviewer: ```suggestion
The Chain ID is the SHA256 hash of the genesis state of the chain, represented as lower case hexadecimal number of 64 digits.
```

#### Reference Definition

The definition is delegated to the [EOSIO Transactions Protocol Documentation](https://developers.eos.io/welcome/v2.0/protocol/transactions_protocol/#32-sign-transaction), the [Chain API Plugin Documentation of the EOSIO Developers Manual](https://developers.eos.io/manuals/eos/latest/nodeos/plugins/chain_api_plugin/api-reference/index?query=chain%20id&page=1#operation/get_info) and the pull request that implemented its generation [Chain ID generation implementation pull request](https://github.com/EOSIO/eos/pull/3425). 

> The Chain ID, as defined by EOSIO, is the SHA256 hash of the genesis state of the chain, represented as lower case hexadecimal number of 64 digits. In order to fit the reference restrictions, the format is a 32 character prefix of the Chain ID.

Reviewer: ```suggestion
The Chain ID, as defined by EOSIO, is the SHA256 hash of the genesis state of the chain, represented as lower case hexadecimal number of 64 digits. In order to fit the CAIP-2 reference format, a 32 character prefix of the Chain ID is used.
```
